client.go

```diff
@@ -53,6 +66,40 @@ func Timeout(timeout time.Duration) func(*Client) error {
 	}
 }
 
+// Keepalive keeps the connection open.
+func Keepalive() func(*Client) error {
+	return func(c *Client) error {
+		go func(c *Client) {
+			for c.IsConnected() {
+				time.Sleep(keepaliveInterval)
+
+				if err := c.setDeadline(); err != nil {
```

Comment:
The errors are just lost, could we do something better?

Reply:
Do you consider this bad practice? 
The errors are handled and not ignored, I don't see any reason why or even where you would want to receive these errors.

Reply:
What if the write error was a temporary one, currently you abort and assume the connection is closed, which may not be the case.

The other thing to watch is with multiple go routines doing writes how do you deal with cross over of messages assuming " \n" generates a response from the server? Even if it doesn't you could be clearing the other goroutines timeout.

Reply:
> What if the write error was a temporary one, currently you abort and assume the connection is closed, which may not be the case.

That is true. Maybe a channel to receive disconnect events might be an option?

> The other thing to watch is with multiple go routines doing writes how do you deal with cross over of messages assuming " \n" generates a response from the server? Even if it doesn't you could be clearing the other goroutines timeout.

It does not generate a response from the server and should therefore neither overwrite any data nor clear the timeout.
I should probably use the mutex though, that should improve the thread-safety.

Reply:
Now I see why you said that it would clear the Deadline but there still is a timeout that will not be cleared so it's fine.

Reply:
yer this is still racey.

Have you considered using a request processor which reads requests from a channel?
```golang
for {
   select {
   case w := <-work:
       process(w)
   case <-time.After(timeout):
       process(keepalive)
   case <-disconnect:
      return
   }
}
```

Reply:
So `work` would be a `chan string` and process would send data?
This would also mean that keepalive is always enabled (which might be a good idea tbh).